jobs/promptLicenceCreationService.test.ts

```diff
@@ -0,0 +1,116 @@
+import { format, add, startOfISOWeek, endOfISOWeek, subDays } from 'date-fns'
+
+import LicenceStatus from '../server/enumeration/licenceStatus'
+import CaseloadService from '../server/services/caseloadService'
+import Container from '../server/services/container'
+import PrisonerService from '../server/services/prisonerService'
+import PromptLicenceCreationService from './promptLicenceCreationService'
+import LicenceType from '../server/enumeration/licenceType'
+
+jest.mock('../server/services/caseloadService')
+jest.mock('../server/services/prisonerService')
+
+const caseloadService = new CaseloadService(null, null, null) as jest.Mocked<CaseloadService>
+const prisonerService = new PrisonerService(null, null) as jest.Mocked<PrisonerService>
+
+const promptLicenceCreationService = new PromptLicenceCreationService(prisonerService, caseloadService)
+
+describe('prompt licence creation service ', () => {
+  beforeEach(() => {
+    prisonerService.searchPrisonersByReleaseDate = jest.fn()
+    caseloadService.pairNomisRecordsWithDelius = jest.fn()
+    caseloadService.filterOffendersEligibleForLicence = jest.fn()
+    caseloadService.mapOffendersToLicences = jest.fn()
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  const managedCases = [
+    {
+      deliusRecord: { offenderId: 1, offenderManagers: [{ active: true, fromDate: format(new Date(), 'yyyy-MM-dd') }] },
+      nomisRecord: {
+        restrictedPatient: false,
+        prisonId: 'someId',
+        conditionalReleaseDate: format(new Date(), 'yyyy-MM-dd'),
+        status: 'someStatus',
+      },
+      licences: [{ status: LicenceStatus.NOT_STARTED, type: LicenceType.AP }],
+      probationPractitioner: { name: 'jim' },
+    },
```

Comment:
meh... you could have used the same pattern that we talked about last time -  extract out a function so you make the bits of the test your interested in pop out in the test. 

This shared data between tests is a bit gnarly and you can't change it for one test without breaking other tests and it becomes difficult to change

Reply:
test refactored. 
The two test do use the same data though